core/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -312,6 +310,47 @@ class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
       ifExists = ctx.EXISTS != null)
   }
 
+/**
+ * Create a [[ShowTableColumnsCommand]] logical plan.
+ *
+ * Syntax:
+ * {{{
+ *   SHOW COLUMNS (FROM | IN) tableName [(FROM | IN) schemaName];
+ * }}}
+ * Examples:
+ * {{{
+ *   SHOW COLUMNS IN delta.`test_table`
+ *   SHOW COLUMNS IN `test_table` IN `test_database`
+ * }}}
+ */
+  override def visitShowColumns(
+      ctx: ShowColumnsContext): LogicalPlan = withOrigin(ctx) {
+    val spark = SparkSession.active
+    val tableName = visitTableIdentifier(ctx.tableName)
+    val schemaName = Option(ctx.schemaName).map(db => db.getText)
+    val tableIdentifier = schemaName match {
+      case Some(db) =>
+        if (tableName.database.exists(_ != db)) {
+          throw DeltaErrors.showColumnsWithConflictDatabasesError(db, tableName)
+        } else {
+          TableIdentifier(tableName.identifier, Some(db))
+        }
+      case None => tableName
+    }
+
+    DeltaTableIdentifier(spark, tableIdentifier) match {
+      // If `tableIdentifier` is a Delta table, get `tablePath` from `path` or `metadata.location`.
+      // If `tableIdentifier` is a non-Delta table or a view, return `null` to fallback to Spark's
+      // own parser.
+      case Some(id) if id.path.nonEmpty =>
```

Comment:
In a second thought, we can pass `DeltaTableIdentifier` into `ShowTableColumnsCommand` and `ShowTableColumnsCommand` can just call `DeltaTableIdentifier.getDeltaLog` to get the DeltaLog instance. It will look like
```
DeltaTableIdentifier(spark, tableIdentifier).map(id => ShowTableColumnsCommand(id)).orNull
```

Reply:
Adjust `ShowTableColumnsCommand` parameter from `path: String` to `tableID: DeltaTableIdentifier`